recipes/darknet/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+
+class DarknetConan(ConanFile):
+    name = "darknet"
+    license = "YOLO"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://pjreddie.com/darknet/"
+    description = "Darknet is a neural network frameworks written in C"
+    topics = ("darknet", "neural network", "deep learning")
+    settings = "os", "compiler", "build_type", "arch", "arch_build"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "OpenCV": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "OpenCV": False,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _commit(self):
+        url = self.conan_data["sources"][self.version]["url"]
+        return url[url.rfind("/")+1:url.rfind(".")]
+
+    @property
+    def _lib_to_compile(self):
+        if not self.options.shared:
+            return "$(ALIB)"
+        else:
+            return "$(SLIB)"
+
+    @property
+    def _shared_lib_extension(self):
+        if self.settings.os == "Macos":
+            return ".dylib"
+        else:
+            return ".so"
+
+    def _activate_makefile(self, option):
+        makefileName = option.upper()
+        with tools.chdir(self._source_subfolder):
+            if getattr(self.options, option) == True:
+                tools.replace_in_file('Makefile', makefileName + '=0', makefileName + '=1')
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("This library is not compatible with Windows")
```

Comment:
THere is a fork for Windows support: https://github.com/AlexeyAB/darknet

But I dunno about the project update, I mean, if it is synchronized with the original version.

Reply:
I think it's good for now to have only officially supported configurations, at least as a first step

Reply:
I think I will move to the AlexeyAB version after we complete this PR, as it seems to become the upstream (the v4 is on this repo only, and the pjreddie one refer to all the papers for the AlexeyAB version)